cni/network/network_windows.go

```diff
@@ -385,7 +385,10 @@ func getNATInfo(nwCfg *cni.NetworkConfig, ncPrimaryIPIface interface{}, enableSn
 			ncPrimaryIP = ncPrimaryIPIface.(string)
 		}
 
-		natInfo = append(natInfo, []policy.NATInfo{{VirtualIP: ncPrimaryIP, Destinations: []string{networkutils.AzureDNS}}, {Destinations: []string{networkutils.AzureIMDS}}}...)
+		// only append natInfo if ncPrimaryIP is ipv4
+		if net.ParseIP(ncPrimaryIP).To4() != nil {
```

Comment:
snat required for dns and imds. isn't there a scenario where user can use ipv6 dns ip (ipv6 equivalent of 168.63.129.16)?

Reply:
I do not think we need nat for ipv6